packages/legend-query/src/components/QueryBuilderValueSpecificationEditor.tsx

```diff
@@ -605,11 +1159,22 @@ export const DateInstanceValueEditor = observer(
 
 export const QueryBuilderValueSpecificationEditor: React.FC<{
```

Comment:
I think the reason this editor is a bit chunky to code in is because we are editing value specification directly.
I think what we should do is built a ValueSpecificationEditorState which is build in the lambda processing and mirrors the rest of query builder app. 
@akphi thoughts?